packages/orchestration/src/orchestration-chat-completion-stream.ts

```diff
@@ -0,0 +1,179 @@
+import { createLogger } from '@sap-cloud-sdk/util';
+import { SseStream } from '@sap-ai-sdk/core';
```

Comment:
this file is mostly code duplication, and since this is not trivial code it would be nice to reduce it. I know the generics are annoying, but maybe you can find a way to represent them without nesting them deeper?

E.g. something alone those lines:
```
static async *_processChunk<T extends SseStream<T>>(
    stream: T
  ): AsyncGenerator<T> {
```

Reply:
I suppose this could be done for most process functions, but not for 
`_processFinishReason()`, as this method accesses different properties based on the chunk type